scipy/special/boost_special_functions.h

```diff
@@ -66,19 +66,37 @@ boost::math::policies::user_overflow_error(const char* function, const char* mes
     return 0;
 }
 
+// Simple wrapper to avoid C++ exceptions from crashing Python
+// Should be used for all functions that do not have special error handling
+template<typename Real, typename Callable>
+Real error_handler(Callable&& func) {
+    try {
+        return func();
+    } catch (const std::underflow_error&) {
+        /* Boost underflow - return 0 */
+        return Real(0);
+    } catch (const std::overflow_error&) {
+        /* Boost overflow - return infinity */
+        return std::numeric_limits<Real>::infinity();
```

Comment:
How does boost signal overflows to a negative infinity?

Reply:
Good point, will investigate.